code/game/objects/items/control_wand.dm

```diff
@@ -155,7 +155,78 @@
 	busy = FALSE
 	icon_state = "hacktool"
 
+/// How long before you can "jangle" your keyring again (to prevent spam)
+#define JANGLE_COOLDOWN 10 SECONDS
+
+/obj/item/door_remote/janikeyring
+	name = "janitor's keyring"
+	desc = "An absolutely unwieldy set of keys attached to a metal ring."
+	icon_state = "keyring"
+	item_state = "keyring"
+	var/busy = FALSE
+	var/cooldown = 0
+	var/hack_speed
+	additional_access = list(ACCESS_MEDICAL, ACCESS_RESEARCH, ACCESS_CONSTRUCTION, ACCESS_CARGO, ACCESS_MINING, ACCESS_KITCHEN, ACCESS_BAR, ACCESS_JANITOR, ACCESS_CHAPEL_OFFICE)
+
+/obj/item/door_remote/janikeyring/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>This keyring has access to Medbay, Science, Engineering, Cargo, the Bar and the Kitchen!</span>"
+
+/obj/item/door_remote/janikeyring/attack_self(mob/user)
+	if(cooldown > world.time)
+		return
+	to_chat(user, "<span class='warning'>You shake [src]!</span>")
+	playsound(src, 'sound/items/keyring_shake.ogg', 50)
+	cooldown = world.time + JANGLE_COOLDOWN
+
+/obj/item/door_remote/janikeyring/afterattack(obj/machinery/door/airlock/D, mob/user, proximity)
+	if(!proximity)
+		return
+	if(!istype(D))
+		return
+	if(busy)
+		to_chat(user, "<span class='warning'>You are already using [src] on the airlock's access panel!</span>")
+		return
+	busy = TRUE
+	to_chat(user, "<span class='notice'>You fiddle with [src], trying different keys to open the [D] airlock...</span>")
```

Comment:
It feels like you refer to the airlocks in a few different ways (`the airlock`, `the [D] airlock, `[D]`). Maybe worth standardizing?